### PR TITLE
Bedrock - HSM updates

### PR DESCRIPTION
**Pull request by tippmar-nr**
Disable AI Monitoring when HSM is enabled. Updates tests accordingly.

### CODE REVIEW

File: tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs [hunk 1]
```diff
@@ -3742,11 +3744,6 @@ public void AiMonitoringStreamingDisabledByEnvironmentVariableWhenAiMonitoringEn
             Assert.That(_defaultConfig.AiMonitoringStreamingEnabled, Is.False);
         }
 
-        [Test]
```
Comment (chynesNR) [inline tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs:3745]: Isn't this still a valid test? Though now that I look at the spec again, isn't it supposed to be enabled by default?

Comment (tippmar-nr) [reply tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs:3745]: This test was essentially a duplicate of `AiMonitoringRecordContentDisabledWhenAiMonitoringDisabled`

**Review by nrcventura — CHANGES_REQUESTED** [review]
We need to review the latest spec changes regarding this before we can merge this PR.

**Review by nr-ahemsath — APPROVED** [review]
Looks solid! 👍

**Review by jaffinito — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Pull request merged**